integration/spark/spark3/src/main/java/io/openlineage/spark3/agent/lifecycle/plan/DataSourceV2RelationInputDatasetBuilder.java

```diff
@@ -43,6 +44,11 @@ public boolean isDefinedAtLogicalPlan(LogicalPlan logicalPlan) {
 
   @Override
   public List<OpenLineage.InputDataset> apply(DataSourceV2Relation relation) {
+    if (context.getSparkExtensionVisitorWrapper() != null
```

Comment:
I don't think null-check is necessary here , If there's a case when wrapper is not initialized, then this case should be fixed ;)